src/test/e2e/manage/roles/roles-overview.admin.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { faker } from '@faker-js/faker';
+import { expect, test } from '@playwright/test';
+
+test.describe('Roles Overview - Admin', () => {
+  test.beforeEach(async ({ page, context }) => {
+    await context.addCookies([
+      {
+        name: 'ff_react_roles_overview',
+        value: 'true',
+        domain: 'localhost',
+        path: '/',
+      },
+    ]);
+    await page.goto('/manage/roles');
+  });
+
+  test('can navigate to roles overview page', async ({ page }) => {
+    await expect(page.getByRole('heading')).toContainText('Rollen');
+  });
+
+  test('can search for roles', async ({ page }) => {
+    const initialRows = page.getByRole('row');
+    const initialFirstRow = await initialRows.nth(1).textContent();
+
+    await page.getByPlaceholder(/zoek/i).fill('e2e');
+    await page.waitForLoadState('networkidle');
+    await expect(page.getByPlaceholder(/zoek/i)).toHaveValue('e2e');
+    const e2eRows = page.getByRole('row');
+    const e2eResultRow = e2eRows.nth(1).filter({
+      hasText: 'e2e',
+    });
+    await expect(e2eResultRow).toBeVisible({ timeout: 8_000 });
+    // Search shouldn't happen when less than 3 characters are typed.
+    await page.getByPlaceholder(/zoek/i).fill('ie');
+
+    const stillE2eRows = page.getByRole('row');
+    const stillE2EResultRow = stillE2eRows.nth(1).filter({
+      hasText: 'e2e',
+    });
+    await expect(stillE2EResultRow).toBeVisible({ timeout: 8_000 });
+
+    await page.getByPlaceholder(/zoek/i).fill(faker.lorem.words(10));
+    await page.waitForLoadState('networkidle');
+    await expect(page.getByText('Geen rollen gevonden.')).toBeVisible();
+
+    await page.getByPlaceholder(/zoek/i).fill('');
+    await page.waitForLoadState('networkidle');
+    await expect(page.getByText('Geen rollen gevonden.')).not.toBeVisible();
+    const resetRows = page.getByRole('row');
+    const resetFirstRow = resetRows.nth(1);
+    await expect(resetFirstRow).toHaveText(initialFirstRow);
+  });
+
+  test('can paginate roles', async ({ page }) => {
+    await page.getByPlaceholder(/zoek/i).fill('eer');
+    await page.waitForTimeout(300);
+    await page.waitForLoadState('networkidle');
+
+    const firstPageRows = page.getByRole('row');
+    const firstPageFirstRow = await firstPageRows.nth(1).textContent();
+
+    await expect(page.getByRole('button', { name: /^2$/ })).toBeVisible();
```

Comment:
Is it possible to avoid regexes?

```suggestion
    await expect(page.getByRole('button', { name: '2'})).toBeVisible();
```